js/ui/collection/ui.collection_widget.edit.js

```diff
@@ -215,37 +214,34 @@ const CollectionWidget = BaseCollectionWidget.inherit({
             filter: that._getCombinedFilter.bind(that),
             totalCount: function() {
                 const items = that.option('items');
-                const dataSource = that._dataSource;
-                return dataSource && dataSource.totalCount() >= 0
-                    ? dataSource.totalCount()
+                const dataController = that._dataController;
+                return dataController.totalCount() >= 0
```

Comment:
let's remember the result of `dataController.totalCount()` to not call function two times

Reply:
done